lxd/storage_lvm.go

```diff
@@ -238,23 +238,27 @@ func (s *storageLvm) StoragePoolInit(config map[string]interface{}) (storage, er
 	}
 
 	source := s.pool.Config["source"]
-	s.vgName = s.pool.Config["lvm.vg_name"]
 	s.thinPoolName = s.pool.Config["lvm.thinpool_name"]
+	if s.pool.Config["lvm.vg_name"] != "" {
+		s.vgName = s.pool.Config["lvm.vg_name"]
+	}
 	if source == "" {
-		// Source property is empty, so the user wants us to reuse an
-		// already existing volume group.
-		s.log.Debug(fmt.Sprintf("Source property of this lvm storage pool empty: Checking that volume group \"lvm.vg_name=%s\" already exists.", s.vgName))
-		ok, err := storageVGExists(s.vgName)
-		if err != nil {
-			// Internal error.
-			return s, err
-		} else if !ok {
-			// Volume group does not exist.
-			return s, fmt.Errorf("The \"source\" property of the storage pool is empty and the \"lvm.vg_name=%s\" volume group does not exist.", s.vgName)
+		return s, fmt.Errorf("Loop backed lvm storage pools are not supported.")
+	} else {
+		if filepath.IsAbs(source) {
+			if !shared.IsBlockdevPath(source) {
+				return s, fmt.Errorf("Loop backed lvm storage pools are not supported.")
+			}
+		} else {
+			ok, err := storageVGExists(source)
+			if err != nil {
+				// Internal error.
+				return s, err
+			} else if !ok {
+				// Volume group does not exist.
+				return s, fmt.Errorf("The requested volume group \"%s\" does not exist.", source)
+			}
 		}
-		s.log.Debug(fmt.Sprintf("Source property of this lvm storage pool empty: Detected that volume group \"lvm.vg_name=%s\" already exists.", s.vgName))
-	} else if filepath.IsAbs(source) && !shared.IsBlockdevPath(source) {
-		return s, fmt.Errorf("Loop backed lvm storage volumes are currently not supported.")
```

Comment:
@brauner yeah, not saying we need to rush it, but it sounds pretty easy to do considering that the code is already there for btrfs and it'd make our testsuite quite a bit cleaner too.